blocksync/reactor.go

```diff
@@ -68,17 +68,35 @@ type Reactor struct {
 func NewReactor(state sm.State, blockExec *sm.BlockExecutor, store *store.BlockStore,
 	blockSync bool, metrics *Metrics,
 ) *Reactor {
-	if state.LastBlockHeight != store.Height() {
-		panic(fmt.Sprintf("state (%v) and store (%v) height mismatch", state.LastBlockHeight,
-			store.Height()))
-	}
 
+	storeHeight := store.Height()
+	var err error
+	if storeHeight == 0 {
+		// If state sync was performed offline and the stores were bootstrapped to height H
+		// the state stores lastHeight will be H while blockstores Height and Base are still 0
+		// 1. This scenario should not lead to a panic in this case, which is indicated by
+		// having a OfflineStateSyncHeight > 0
+		// 2. We need to instruct the blocksync reactor to start fetching blocks from H+1
+		// instead of 0.
+		storeHeight, err = blockExec.Store().GetOfflineStateSyncHeight()
+		if err != nil && err.Error() != "value empty" {
+			panic(fmt.Sprintf("failed to retrieve statesynced height from store %s", err))
```

Comment:
Could you also add `state.LastBlockHeight` somehow in the panic message (since it was appearing before)

Reply:
Following my comment below. Do you think we can write a more expressive message to make sure the operator know what to do if they see this message? (likely, wipe out the data dir and start over)